test/integration/puppetlabs/services/jruby_pool_manager/jruby_pool_int_test.clj

```diff
@@ -376,7 +300,7 @@
 
          (jruby-core/flush-pool! pool-context)
          ; wait until the flush is complete
-         (await (jruby-agents/get-pool-agent pool-context))
+         (is (jruby-testutils/wait-for-instances (jruby-internal/get-pool pool-context) 1))
```

Comment:
Yeah, I need to figure out a better way to wait for the flush to finish. The flush function takes the lock to get the jrubies, then does the actual refilling happens on the agent thread asynchronously. I think we were talking about how making the flush completely synchronous might be a good idea. That would remove the ambiguity in the tests and just make it more predictable in general for consumers

The only weird thing would be that jrubies will be in the pool available for borrowing before the flush completes since it destroys and remakes them one by one. The flush would always finish though, since it just creates a set number of jrubies and doesn't care if some have been borrowed before it finishes